metals/src/main/scala/scala/meta/internal/implementation/SuperMethodProvider.scala

```diff
@@ -1,295 +1,27 @@
 package scala.meta.internal.implementation
 
-import scala.collection.{mutable => m}
-
-import scala.meta.internal.metals.codelenses.SuperMethodCodeLens.LensGoSuperCache
-import scala.meta.internal.semanticdb.ClassSignature
-import scala.meta.internal.semanticdb.MethodSignature
-import scala.meta.internal.semanticdb.Scala._
 import scala.meta.internal.semanticdb.SymbolInformation
-import scala.meta.internal.semanticdb.SymbolOccurrence
-import scala.meta.internal.semanticdb.TextDocument
-import scala.meta.internal.semanticdb.TypeRef
-import scala.meta.internal.semanticdb.TypeSignature
 
 object SuperMethodProvider {
 
   def findSuperForMethodOrField(
-      methodSymbolInformation: SymbolInformation,
-      documentWithPath: TextDocumentWithPath,
-      symbolRole: SymbolOccurrence.Role,
-      findSymbol: String => Option[SymbolInformation],
-      cache: LensGoSuperCache
+      methodSymbolInformation: SymbolInformation
   ): Option[String] = {
-    for {
-      methodSignature <- isDefinitionOfMethodField(
-        symbolRole,
-        methodSymbolInformation
-      )
-      superSymbol <- findSuperForMethodOrFieldChecked(
-        methodSymbolInformation,
-        methodSignature,
-        documentWithPath,
-        cache,
-        findSymbol
-      )
-    } yield superSymbol
+    getSuperMethodHierarchy(
+      methodSymbolInformation
+    ).headOption
```

Comment:
One concern is that the order of `overriddenSymbol` is semanticdb extractor implementation-dependent.
At least Scala2 (and Scala3 will) returns symbol in order of direct parent -> grand-parent ..., but to be robust to the change in Scala2 and Scala3, we may should check the direct super class of the `SymbolInformation` and pick the symbol from the parent class from `overriddenSymbol`.